packages/core/root/core-root-server-internal/src/events/kibana_started.ts

```diff
@@ -0,0 +1,175 @@
+  constructor: UptimePerStep;
+  preboot: UptimePerStep;
+  setup: UptimePerStep;
+  start: UptimePerStep;
+  elasticsearch: {
+    waitTime: number;
+  };
+  savedObjects: {
+    migrationTime: number;
+  };
+}
+
+export const registerKibanaStartedEvent = (analytics: AnalyticsServiceSetup) => {
+  analytics.registerEventType<{ uptime_per_step: UptimeSteps }>({
+    eventType: KIBANA_STARTED_EVENT_NAME,
+    schema: {
+      uptime_per_step: {
+        properties: {
+          constructor: {
+            properties: {
+              start: {
+                type: 'float',
+                _meta: {
+                  description:
+                    'Number of milliseconds the Node.js process has been running until the constructor was called',
+                },
+              },
+              end: {
+                type: 'float',
+                _meta: {
+                  description:
+                    'Number of milliseconds the Node.js process has been running until the constructor finished',
+                },
+              },
+            },
+          },
+          preboot: {
+            properties: {
+              start: {
+                type: 'float',
+                _meta: {
+                  description:
+                    'Number of milliseconds the Node.js process has been running until `preboot` was called',
+                },
+              },
+              end: {
+                type: 'float',
+                _meta: {
+                  description:
+                    'Number of milliseconds the Node.js process has been running until `preboot` finished',
+                },
+              },
+            },
+          },
+          setup: {
+            properties: {
+              start: {
+                type: 'float',
+                _meta: {
+                  description:
+                    'Number of milliseconds the Node.js process has been running until `setup` was called',
+                },
+              },
+              end: {
+                type: 'float',
+                _meta: {
+                  description:
+                    'Number of milliseconds the Node.js process has been running until `setup` finished',
+                },
+              },
+            },
+          },
+          start: {
+            properties: {
+              start: {
+                type: 'float',
+                _meta: {
+                  description:
+                    'Number of milliseconds the Node.js process has been running until `start` was called',
+                },
+              },
+              end: {
+                type: 'float',
+                _meta: {
+                  description:
+                    'Number of milliseconds the Node.js process has been running until `start` finished',
+                },
+              },
+            },
+          },
+          elasticsearch: {
+            properties: {
+              waitTime: {
+                type: 'float',
```

Comment:
nit: this is probably a legacy issue (and they don't make much difference here)... but I think they are `long` :)